src/object/class_object.c

```diff
@@ -4069,6 +4069,33 @@ classobj_find_constraint_by_attrs (SM_CLASS_CONSTRAINT * cons_list, DB_CONSTRAIN
   const char **namep;
   int i, len, order;
 
+  /* for foreign key, need to check redundancy first */
+  if (new_cons == DB_CONSTRAINT_FOREIGN_KEY)
+    {
+      for (cons = cons_list; cons; cons = cons->next)
+	{
+	  /* check foreign key only */
+	  if (!cons->attributes || cons->type != SM_CONSTRAINT_FOREIGN_KEY)
```

Comment:
How about checking only cons->type?

Reply:
I'd like to keep the code because **cons->attributes** is checked first.